gh-data.py

```diff
@@ -0,0 +1,159 @@
+
+def process_labels(labels):
+    position = None
+    venues = []
+    concerns = []
+    topics = []
+
+    for label in labels:
+        # Position
+        if label["name"] == "blocked":
+            assert position is None
+            position = "blocked"
+        elif label["name"].startswith("position: "):
+            assert position is None
+            position = label["name"][len("position: ") :]
+        # Venue
+        elif label["name"].startswith("venue: "):
+            venues.append(label["name"][len("venue: ") :])
+        # Concerns
+        elif label["name"].startswith("concerns: "):
+            concerns.append(label["name"][len("concerns: ") :])
+        # Topics
+        elif label["name"].startswith("topic: "):
+            topics.append(label["name"][len("topic: ") :])
+
+    return {
+        "position": position,
+        "venues": venues,
+        "concerns": concerns,
+        "topics": topics,
+    }
+
+
+def process_body(issue):
+    lines = issue["body"].splitlines()
+
+    body = {
+        "title": None,
+        "url": None,
+        "explainer": None,
+        "caniuse": None,
+        "bug": None,
+        "webkit": None,
+    }
+
+    mapping = {
+        "specification title": "title",
+        "specification or proposal url (if available)": "url",
+        "specification or proposal url": "url",
+        "explainer url (if available)": "explainer",
+        "explainer url": "explainer",
+        "caniuse.com url (optional)": "caniuse",
+        "caniuse.com url": "caniuse",
+        "bugzilla url (optional)": "bug",
+        "bugzilla url": "bug",
+        "webkit standards-position": "webkit",
+    }
+
+
+    for line in lines:
+        if line == "### Other information":
+            break
+        for title, key in mapping.items():
+            text_title = f"* {title}: "
+            if line.lower().startswith(text_title):
+                value = line[len(text_title) :].strip()
+                value = re.sub(r"\[[^\]]+\]\(([^\)]+)\)", r"\1", value)
+                if value != "" and value.lower() != "n/a":
+                    body[key] = value
+                break
+    return body
+
+
+# Setup
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-u",
+        "--update",
+        action="store_true",
+        help="get the latest issue data from GitHub",
+    )
+    parser.add_argument("-p", "--process", action="store_true", help="process the data")
+    args = parser.parse_args()
+
+    if args.update:
+        # GitHub allows us to read issues in increments of 100, called pages.
+        data = []
+        page = 1
+        while True:
+            try:
+                response = requests.get(
+                    f"https://api.github.com/repos/mozilla/standards-positions/issues?direction=asc&state=all&per_page=100&page={page}",
+                    timeout=5,
+                )
+                response.raise_for_status()
+            except Exception:
```

Comment:
It might pay to check for rate limits.  We have a fair number of issues and I have first-hand experience with hitting those limits.  Look at [this code](https://github.com/MikeBishop/archive-repo/blob/2683167304c4a14f11ee370b535fc23620044a6a/archive-repo/archive.py#L459-L482) for some clues on how to manage that.

Also, this request is not using `$GITHUB_TOKEN`, but I've found that doing that helps a fair bit.  It means setting a [header](https://github.com/MikeBishop/archive-repo/blob/2683167304c4a14f11ee370b535fc23620044a6a/archive-repo/archive.py#L730-L733), but it means that you are less likely to hit rate limits.

Reply:
https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users says 60 requests per hour, and currently this will make 12 requests. (Could make it avoid the last request by looking at the `link` response header.)

If we make a GH Action workflow to batch updates if the most recent run was less than 15 minutes ago or so (which seems like a good idea anyway), unauthenticated shouldn't hit the rate limit for now.

Reply:
I've experimented in https://github.com/zcorpan/gh-actions-test to get such a setup working (with help of ChatGPT 4o). The workflow will push built files to `gh-pages`, which I think is useful to be able to see the history.